src/test/kotlin/org/elm/ide/inspections/TypeInferenceInspectionTest.kt

```diff
@@ -185,6 +184,30 @@ main : R
 main = foo <error descr="Type mismatch.Required: { r | x : () }Found: { y : () }">{ y = () }</error>
 """)
 
+
+    fun `test extension record argument with more fields than extension parameter`() = checkByText("""
```

Comment:
Note that these test cases are different: in the first, `Large a` is being assigned to `Small a`. In the second, `Small a` is being assigned to `Large a`.